source/include/defender.h

```diff
@@ -177,112 +245,364 @@ typedef enum
+ */
 #define DEFENDER_REPORT_LENGTH_PORTS_KEY                      STRING_LITERAL_LENGTH( DEFENDER_REPORT_PORTS_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "port" key in the defender report.
+ */
 #define DEFENDER_REPORT_PORT_KEY                              DEFENDER_REPORT_SELECT_KEY( "port", "pt" )
+
+/**
+ * @ingroup defender_constants
+ * @brief Length of the "port" key in the defender report.
+ */
 #define DEFENDER_REPORT_LENGTH_PORT_KEY                       STRING_LITERAL_LENGTH( DEFENDER_REPORT_PORT_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "interface" key in the defender report.
+ */
 #define DEFENDER_REPORT_INTERFACE_KEY                         DEFENDER_REPORT_SELECT_KEY( "interface", "if" )
+
+/**
+ * @ingroup defender_constants
+ * @brief Length of the "interface" key in the defender report.
+ */
 #define DEFENDER_REPORT_LENGTH_INTERFACE_KEY                  STRING_LITERAL_LENGTH( DEFENDER_REPORT_INTERFACE_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "listening_udp_ports" key in the defender report.
+ */
 #define DEFENDER_REPORT_UDP_LISTENING_PORTS_KEY               DEFENDER_REPORT_SELECT_KEY( "listening_udp_ports", "up" )
+
+/**
+ * @ingroup defender_constants
+ * @brief Length of the "listening_udp_ports" key in the defender report.
+ */
 #define DEFENDER_REPORT_LENGTH_UDP_LISTENING_PORTS_KEY        STRING_LITERAL_LENGTH( DEFENDER_REPORT_UDP_LISTENING_PORTS_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "network_stats" key in the defender report.
+ */
 #define DEFENDER_REPORT_NETWORK_STATS_KEY                     DEFENDER_REPORT_SELECT_KEY( "network_stats", "ns" )
+
+/**
+ * @ingroup defender_constants
+ * @brief Length of the "network_stats" key in the defender report.
+ */
 #define DEFENDER_REPORT_LENGTH_NETWORK_STATS_KEY              STRING_LITERAL_LENGTH( DEFENDER_REPORT_NETWORK_STATS_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "bytes_in" key in the defender report.
+ */
 #define DEFENDER_REPORT_BYTES_IN_KEY                          DEFENDER_REPORT_SELECT_KEY( "bytes_in", "bi" )
+
+/**
+ * @ingroup defender_constants
+ * @brief Length of the "bytes_in" key in the defender report.
+ */
 #define DEFENDER_REPORT_LENGTH_BYTES_IN_KEY                   STRING_LITERAL_LENGTH( DEFENDER_REPORT_BYTES_IN_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "bytes_out" key in the defender report.
+ */
 #define DEFENDER_REPORT_BYTES_OUT_KEY                         DEFENDER_REPORT_SELECT_KEY( "bytes_out", "bo" )
+
+/**
+ * @ingroup defender_constants
+ * @brief Length of the "bytes_out" key in the defender report.
+ */
 #define DEFENDER_REPORT_LENGTH_BYTES_OUT_KEY                  STRING_LITERAL_LENGTH( DEFENDER_REPORT_BYTES_OUT_KEY )
 
+/**
+ * @ingroup defender_constants
+ * @brief "packets_in" key in the defender report.
+ */
 #define DEFENDER_REPORT_PKTS_IN_KEY                           DEFENDER_REPORT_SELECT_KEY( "packets_in", "pi" )
```

Comment:
I feel weird about this packet abbreviation because none of the other macros are abbreviated. But it's fine.

Reply:
This is needed to address MISRA violation so that an identifier is different in first 31 characters.